parser/est/est.go

```diff
@@ -54,8 +54,9 @@ type Service struct {
 }
 
 type Param struct {
-	IsPtr bool
-	Decl  *schema.Decl
+	IsPtr         bool
+	Decl          *schema.Decl
+	TypeArguments []*schema.Type // Any type arguments on the declaration (nil if none)
```

Comment:
I was tempted to simply swap this over for:

```suggestion
	Type          *schema.Type
```

But this struct is used in a lot of places, so I'll do that as a follow on once everywhere else is updated to support type arguments (i.e. this way I get to implement this piece by piece without having to touch everything at once)